### PR TITLE
Selection Checkboxes for Pathways List

### PR DESCRIPTION
**Pull request by radamson**
Adds selection checkboxes for the pathways in the pathways list.  Pathways can then be deleted and exported as a group.

Potentially obnoxious gifs of it in action below:

**Selection:**

selection

**Export:**

export

**Delete:**

delete

### CODE REVIEW

**Review by noranda — CHANGES_REQUESTED** [review]
Looking good! The select all button works well. Just a few issues:

1. The styling is a bit off, specifically the ripple circle is more oval. To fix:
Screen Shot 2020-10-19 at 11 16 38 AM

2. The action buttons (export, move to, delete) should only render if one or more checkboxes are selected, otherwise there are no actions to do (see design). Currently, if you try to export with nothing selected, you will export an empty zip. 

3. We should also include checkboxes in the Criteria List. Is this a separate task, or is it easy enough to add here?

File: src/components/PathwaysList/PathwaysListCheckbox.tsx
Comment (noranda) [inline src/components/PathwaysList/PathwaysListCheckbox.tsx:1]: Since this is a hook and not a component, and since we will want to reuse it for the criteria list, why don't we create a `src/hooks` folder to keep it in. We could also move the `useRefState` hook from utils into this new folder as well.

File: src/components/PathwaysList/PathwaysList.tsx
Comment (blangley28) [inline src/components/PathwaysList/PathwaysList.tsx:73]: If there's only one file selected do we want to download as just a json file? The logic can then be expanded for referenced pathways so pathways with references are exported as zips but standalone pathways are json

**Review by blangley28 — COMMENTED** [review]
Tested all the functionality out and it works. Good to go after Noranda's comments are incorporated

File: src/components/PathwaysList/PathwaysList.tsx
Comment (noranda) [reply src/components/PathwaysList/PathwaysList.tsx:73]: My vote is yes on this.

**Comment by noranda** [comment]
Looks great! One last comment/question. We should probably have the confirmation popup for when we delete pathways, like we do when you click on the delete button in the table. It might say "Are you sure that you would like to delete the selected pathway(s)?" If it's easy enough to add to this PR?

**Comment by radamson** [comment]
I think all the comments up to this point have now been addressed.  As I was working through this I noticed that the CriteriaList and PathwayList components share a lot of the same code.  I started working on de-duplicating some of this code into some shared components, but it wasn't easy enough for me to finish just yet.  I think any code that refactors these components would be better suited as a separate PR for ease of review so I'll hold off on commit anything for that unless we decide we'd rather tackle that in this PR.

File: src/utils/builder.ts
Comment (blangley28) [inline src/utils/builder.ts:46]: Is there a reason you use `([] as Pathway[]).concat(pathway)[0]` instead of just `pathway`. Seems like a very long way to get back what you started with....

It might also just be easier to make `pathway` always be a list and then just check the length in the if statement below.

**Review by blangley28 — COMMENTED** [review]
Looks good, I like the new hook directory. I agree those changes should be put up in a separate PR just so its easier to review.

File: src/utils/builder.ts
Comment (radamson) [reply src/utils/builder.ts:46]: `([] as Pathway[]).concat(pathway)[0]` is used in the case where only a single pathway and wraps that pathway in an array.  I'd be fine with making pathway always be a list, but other existing code already expects to be able to pass in a single pathway and might need similar checks anyways.

**Review by noranda — APPROVED** [review]

**Review by blangley28 — APPROVED** [review]

**Pull request merged**
